changes.go

```diff
@@ -0,0 +1,207 @@
+package apidApigeeSync
+
+import (
+	"github.com/30x/apid-core"
+	"time"
+	"net/url"
+	"path"
+	"net/http"
+	"io/ioutil"
+	"encoding/json"
+
+	"github.com/apigee-labs/transicator/common"
+
+)
+
+var lastSequence string
+var block        string = "45"
+
+/*
+ * Long polls the change agent with a 45 second block. Parses the response from
+ * change agent and raises an event. Called by pollWithRetry().
+ */
+func pollChangeAgent(quit chan bool) error {
+
+	changesUri, err := url.Parse(config.GetString(configChangeServerBaseURI))
+	if err != nil {
+		log.Errorf("bad url value for config %s: %s", changesUri, err)
+		return err
+	}
+	changesUri.Path = path.Join(changesUri.Path, "changes")
+
+	/*
+	 * Check to see if we have lastSequence already saved in the DB,
+	 * in which case, it has to be used to prevent re-reading same data
+	 */
+	lastSequence = getLastSequence()
+
+	for {
+		select {
```

Comment:
This is a good point, but in this suggested code, if the quit signal is received, the `getChanges` goroutine will still be running.  The purpose of this quit is to ensure that no goroutines are left lying around after `pollWithBackoff` returns so that you can be sure there are no goroutines still running which could modify global state, and thus impair testing.  Also, when we implement a command/control channel, this might be a guarantee we need before sending a confirmation back.  For instance, you send a "stop polling" command.  By the time that we return a response, we should guarantee there aren't any routines/processes running in that capacity